Library/Homebrew/cmd/outdated.rb

```diff
@@ -40,11 +49,20 @@ def print_outdated(formulae)
 
   def print_outdated_json(formulae)
     json = []
-    outdated = formulae.select(&:outdated?).each do |f|
+    fetch_head = ARGV.fetch_head?
+    outdated_formulae = formulae.select { |f| f.outdated?(:fetch_head => fetch_head) }
+
+    outdated = outdated_formulae.each do |f|
+      outdated_versions = f.outdated_versions(:fetch_head => fetch_head)
+      current_version = if f.tab_version_outdated?(f.latest_head_version)
```

Comment:
It's bad practice to use `f.latest_head_version`, because it's only referred to one of version and has different semantical meaning with `outdated_versions`.

This can(should) be `if f.head? && outdated_versions.include?(f.pkg_version)`. The idea is that:
(1) You should check whether it's a head formula.
(2) If `f.pkg_version` is the same as one of installed versions, it means that you didn't fetch latest HEAD.

However, in order to make it work, you are required to implement `HeadVersion#==`.